altos-rust/port/cortex-m0/src/interrupt/defs.rs

```diff
@@ -0,0 +1,81 @@
+// interrupt/defs.rs
+// AltOSRust
+//
+// Created by Mohammed Inoue on 2/21/17
+
+
+#[derive(Copy, Clone)]
+pub enum Hardware {
+    WWDG,
+    PVDVDDIO2,
+    RTC,
+    FLASH,
+    RCCCRS,
+    EXTI01,
+    EXTI23,
+    EXTI415,
+    TSC,
+    DMACH1,
+    DMACH23,
+    DMACH4PLUS,
+    ADCCOMP,
+    TIM1BRKUP,
+    TIM1CC,
+    TIM2,
+    TIM3,
+    TIM6,
+    TIM7,
+    TIM14,
+    TIM15,
+    TIM16,
+    TIM17,
+    I2C1,
+    I2C2,
+    SPI1,
+    SPI2,
+    USART1,
+    USART2,
+    USART3PLUS,
+    CECCAN,
+    USB,
+}
+
+pub fn map_hardware(target:Hardware) -> u8 {
+    match target {
```

Comment:
The rust compiler got mad at me for doing that since you try to multiple and divide those numbers later in functions within priority.rs. So I used the mapping function, so I know its a u8, unless there is a way to specify when making an enum that they are a specific type or something.